File: String/Cpp/1-Valid_Palindrome.cpp
// Question Link -> "https://leetcode.com/problems/valid-palindrome/description/"

#include<bits/stdc++.h>
using namespace std;

bool validPalindrome(string input){
    if(input == " "){
        return true;
    }
    string ans;
    
    // first checking each character to eliminate non alpha-numeric characters & space and converting to lower case 
    for(char c : input){
        if(isalnum(c)){
            ans += tolower(c);
        }
    }
    
    // taking 2 pointers to compare each character from starting and ending to check palindrome
    int i = 0, j = ans.length()-1;
    
    while(i<j){
        if(ans[i] != ans[j]){
            return false;
        }
        i++;
        j--;
    }
    
    return true;
}


int main(){
    
    // TAKING INPUT STRING
    string input;
    cout << "Enter Input String : ";
    cin >> input;
    
    // CALLING FUNCTION
    if(validPalindrome(input)){
        cout << "true" << endl;
    }
    else cout << "false" << endl;
    
}

File: String/Python/1-Valid_Palindrome.py
# Question Link -> "https://leetcode.com/problems/valid-palindrome/description/"

def validPalindrome(string):
    if len(string) == 0 or len(string) == 1:
        return True
        
    ans = ''
    
    # first checking each char to elinimate non alphanumeric character & spaces and converting to lower case
    for char in string:
        if char.isalnum():
            ans += char.lower()
         
    # taking 2 pointers for comparing each character from start and end to check string is palindrome   
    i = 0
    j = len(ans)-1
    
    while(i < j):
        if ans[i] != ans[j]:
            return False
        i += 1
        j -= 1
        
    return True
    

string = input("Enter Input string : ")
print(validPalindrome(string))
